homeassistant/components/history/__init__.py

```diff
@@ -124,6 +124,7 @@ class LazyState(history_models.LazyState):
         vol.Required("start_time"): str,
         vol.Optional("end_time"): str,
         vol.Optional("statistic_ids"): [str],
+        vol.Optional("period"): str,
```

Comment:
Not sure what we should accept here, the implementation currently only understands `"hour"`, which will fetch long term statistics, everything else will fetch 5-minute statistics.

Reply:
It should default to hour, as it should be backwards compatible?

Reply:
Yeah, and it does default to `"hour"` in the implementation, but how should frontend specify the period?
Magic strings? A number of seconds or minutes?

Reply:
magic strings

Reply:
Either constants:

```suggestion
        vol.Optional("period"): vol.Any("hour", "5minute"),
```

Or we allow specifying in minutes
```suggestion
        vol.Optional("period"): vol.Any(5, 60),
```

Reply:
Should this also be required?

Reply:
Constants added in f716ede